### DISCUSSION TITLE
Source code

### DISCUSSION THREAD

**Discussion by mattkenig** [elliotwaite/timestream#1] [General]
Hi - I am having trouble finding the source code. I only see the images in the repo. Can you advise? thanks

**Comment by elliotwaite** [comment]
Hi, sorry for the confusion. Timestream currently isn't an open-source project so this repo doesn't contain the app's source code.  I just added this About section to the README to help explain this:

> ### About
> Welcome to Timestream's GitHub repo. This repo is currently only used for tracking [issues](https://github.com/elliotwaite/timestream/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc) and holding [discussions](https://github.com/elliotwaite/timestream/discussions). Timestream isn't an open source project, so this repo doesn't host the app's source code, but in the future this repo may also be used to allow users to contribute code for certain parts of the app, for example, to allow users to contribute custom timer themes or other extension-like features.

Let me know if you have any suggestions for how I could improve this About section.

The reason Timestream isn't open source is because I plan on adding revenue streams to the app in the future. For example, I may have it sometimes display ads at the end of timers, or I may offer a Pro version with extra features. So I didn't think it made sense to open-source the code for a project with this type of business model.

But I'm a fan of GitHub's issue tracker and discussion features and I thought other developers might be familiar with them as well, so I thought it would be a good idea to create a repo that would allow users to submit issues and start discussions using this platform. I also thought that this repo could potentially be used in the future to allow users to contribute code to the app in the form of custom timer themes or other extension-like features. But I understand that public GitHub repos are typically used for open-source projects, so I wasn't sure if using GitHub in the way that I am would be a good idea. What do you think? Is it too confusing or awkward to use GitHub in this way?

**Comment by mattkenig** [comment]
Thanks for the quick reply. No worries.

Your addition to the README helps clarify it.

Keep us updated if things change on your end and you open-source it, or if
you’d be open to someone purchasing the source code.

Thanks

On Thu, Sep 15, 2022 at 3:00 PM Elliot Waite ***@***.***>
wrote:

> Hi, sorry for the confusion. Timestream currently isn't an open-source
> project so this repo doesn't contain the app's source code. I just added
> this About section to the README to help explain this:
>
> About
>
> Welcome to Timestream's GitHub repo. This repo is currently only used for
> tracking issues
> <https://github.com/elliotwaite/timestream/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc>
> and holding discussions
> <https://github.com/elliotwaite/timestream/discussions>. Timestream isn't
> an open source project, so this repo doesn't host the app's source code,
> but in the future this repo may also be used to allow users to contribute
> code for certain parts of the app, for example, to allow users to
> contribute custom timer themes or other extension-like features.
>
> Let me know if you have any suggestions for how I could improve this About
> section.
>
> The reason Timestream isn't open source is because I plan on adding
> revenue streams to the app in the future. For example, I may have it
> sometimes display ads at the end of timers, or I may offer a Pro version
> with extra features. So I didn't think it made sense to open-source the
> code for a project with this type of business model.
>
> But I'm a fan of GitHub's issue tracker and discussion features and I
> thought other developers might be familiar with them as well, so I thought
> it would be a good idea to create a repo that would allow users to submit
> issues and start discussions using this platform. I also thought that this
> repo could potentially be used in the future to allow users to contribute
> code to the app in the form of custom timer themes or other extension-like
> features. But I understand that public GitHub repos are typically used for
> open-source projects, so I wasn't sure if using GitHub in the way that I am
> would be a good idea. What do you think? Is it too confusing or awkward to
> use GitHub in this way?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/elliotwaite/timestream/discussions/1#discussioncomment-3656714>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AOBENYNNIEG3QZUZ4J3QBITV6NW3TANCNFSM6AAAAAAQNGQX6U>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
-- 
Matt Kenig

**Reply by elliotwaite** [reply]
Okay, I'll post something here if this project becomes more open-source in the future.

Also, I'm open to discussing any of the technology used to build and serve the app in case any devs are interested. Right now I'm using React / React Native and NextJS on the server with a Supabase DB. I'm using a monorepo setup (pretty much [this Solito one](https://solito.dev/starter)) to share code between the web app and the native apps.

**Comment by mattkenig** [comment]
Sounds good. Thanks Elliot. Good luck with everything

On Thu, Sep 15, 2022 at 8:51 PM Elliot Waite ***@***.***>
wrote:

> Okay, I'll post something here if this project becomes more open-source in
> the future.
>
> Also, I'm open to discussing any of the technology used to build and serve
> the app in case any devs are interested. Right now I'm using React / React
> Native and NextJS on the server with a Supabase DB. I'm using a monorepo
> setup (pretty much this Solito one <https://solito.dev/starter>) to share
> code between the web app and the native apps.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/elliotwaite/timestream/discussions/1#discussioncomment-3658370>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AOBENYI63QARU3SUMJJVOATV6PACHANCNFSM6AAAAAAQNGQX6U>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
-- 
Matt Kenig

**Reply by elliotwaite** [reply]
Thanks!
